packages/amplify_auth_cognito/lib/src/CognitoSession/AWSCredentials.dart

```diff
@@ -23,18 +23,18 @@ class AWSCredentials {
   String awsSecretKey;
   String sessionToken;
   AWSCredentials.init({@required LinkedHashMap<dynamic, dynamic> creds}) {
-    var realCreds;
-    if (Platform.isAndroid) {
-      if (creds.containsKey("value")) {
-        realCreds = creds["value"];
+    if (creds != null) {
+      var realCreds = {};
+      if (Platform.isAndroid) {
+        if (creds.containsKey("value")) {
+          realCreds = creds["value"];
+        }
       } else {
```

Comment:
Does Dart provide a switch over the various `Platform` values? Either way, does it make sense to explicitly write `else if (Platform.isIOS)`, here?

Reply:
@jamesonwilliams I moved this logic to the platform code.